contexts/PaymentContext/src/Domain/ChecksumGenerator.php

```diff
@@ -30,14 +33,13 @@ public function __construct( array $checksumCharacters ) {
 	 * @return string The checksum as a single character present in the constructors array argument
 	 */
 	public function createChecksum( string $string ): string {
-		$checksum = array_sum(
-			array_map(
-				'ord', // Returns the ASCII value of the character
-				str_split( md5( $string ) )
-			)
-		);
+		$checksum = hexdec( substr( md5( $this->normalizeString( $string ) ), 0, 8 ) );
```

Comment:
Why 8, not - say - 16?

Reply:
8 chars equals 32 bit. 16 chars would be 64 bits, which would break the checksum mechanism on 32 bit PHP platforms where `PHP_INT_SIZE` is 4 (bytes). On those platforms the resulting number from `hexdec` would be a float and modulos of floats are always 0. 

I don't have the necessary math to prove it, but my gut feeling is that using a smaller portion of the hash is not a problem for possible hash collisions since our character set only allows for 20 different outcomes, which could be expressed in 5 bits and we could even use CRC (which returns 8 an bit checksum) instead of MD5 for that.

Reply:
Not saying we should introduce more incompatibilities, but if we really strive to support x64 we [should strengthen our efforts](https://travis-ci.org/wmde/FundraisingFrontend/jobs/278578315#L3700).

Reply:
So this change broke our compat with x64? If so I suggest reverting the relevant part